### PR TITLE
PLAT-382 - convert mds-compliance-engine to Jest

### PR DESCRIPTION
**Pull request by nessur**
## 📚 Purpose
*[Convert all tests in mds-compliance-engine from mocha to jest]*

## 👌 Resolves:
- [x] ✨ PLAT-382 - convert test suites (mds-compliance-engine)

## 📦 Impacts:
*[mds-compliance-engine]*

### CODE REVIEW

File: packages/mds-compliance-engine/package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   ],
   "scripts": {
     "build": "tsc --build tsconfig.build.json",
-    "test": "DOTENV_CONFIG_PATH=../../.env nyc --use-spawn-wrap ts-mocha --project ../../tsconfig.json --exit",
+    "test": "DOTENV_CONFIG_PATH=../../.env jest --runInBand --forceExit",
```
Comment (drtyh2o) [inline packages/mds-compliance-engine/package.json:17]: Is there no resolution for `--forceExit`?

File: packages/mds-compliance-engine/tests/engine/count-tests.spec.ts [hunk 2]
```diff
@@ -119,22 +118,22 @@ describe('Tests Compliance Engine Count Functionality:', () => {
...
+      expect(
         isCountRuleMatch(
           COUNT_POLICY.rules[0] as CountRule,
           GEOGRAPHIES,
           LAdevices[0],
           LAevents[0] as VehicleEvent & { telemetry: Telemetry }
         )
-      )
-      test.assert(
+      ).toBeTruthy()
```
Comment (drtyh2o) [inline packages/mds-compliance-engine/tests/engine/count-tests.spec.ts:128]: Do you prefer `.toBeTruthy()` to `.toEqual(true)`? Are there instances where a `boolean` is not returned?

**Review by drtyh2o — COMMENTED** [review]
Any chance to convert `done => {...; done() }` to `async () => { ...; }` in these test suites? Also, looks like a `before` was missed.
Screen Shot 2022-03-21 at 5 02 34 PM

File: packages/mds-compliance-engine/package.json (hunk 1 above)
Comment (nessur) [reply packages/mds-compliance-engine/package.json:17]: `--detechOpenHandles` doesnt offer any help, is a broken feature, as documented by open Issues on GH.

**Review by avatarneil — APPROVED** [review]

**Comment by nessur** [comment]
> Any chance to convert `done => {...; done() }` to `async () => { ...; }` in these test suites? Also, looks like a `before` was missed.
> Screen Shot 2022-03-21 at 5 02 34 PM

All cleaned up! `done()` are gone.

**Review by drtyh2o — APPROVED** [review]

**Pull request merged**